woocommerce/woocommerce-android
# Product Creation with AI: feedback view

Closes: #9871 

### Description
This PR implements the feedback view in the product creation form, the view is implemented as a reusable component, and it's shared between the product creation form and the product sharing dialog, please check p1696239553815609-slack-C03L1NF1EA3 for discussion around this.

### Testing instructions
1. Use an AI eligible site.
2. Open the app then start AI product creation.
3. When the product preview is generated, confirm a feedback view is shown.
4. Click on either button, and confirm the view is hidden and a tracking event is sent.

### Images/gif
| Light | Dark |
| --- | --- |
| ![Screenshot_20231002_161938](https://github.com/woocommerce/woocommerce-android/assets/1657201/3b74e793-68b3-4201-abaf-30988afd4506) | ![Screenshot_20231002_161951](https://github.com/woocommerce/woocommerce-android/assets/1657201/47c45b97-b32d-4f03-a79f-6da3dc9d6dad) |

##### Product sharing dialog updates

| Before | After |
| --- | --- |
| ![Screenshot_20231002_161212](https://github.com/woocommerce/woocommerce-android/assets/1657201/7375ee1d-1c07-4d5c-816d-68f54fbcb7b5) | ![Screenshot_20231002_161720](https://github.com/woocommerce/woocommerce-android/assets/1657201/d3e3fa18-bc5e-438c-b315-cc89f7eecaac) |
| ![Screenshot_20231002_161227](https://github.com/woocommerce/woocommerce-android/assets/1657201/ae06b8ed-2fb7-47da-adac-3afa9c5a5765) | ![Screenshot_20231002_161703](https://github.com/woocommerce/woocommerce-android/assets/1657201/6251f158-5dd1-4aaf-95eb-a72df8f8ebe5) |
